src/rust/uk.mangainua/src/lib.rs

```diff
@@ -0,0 +1,380 @@
+			let url = format!("{}/{}", base_url, status_value);
+			
+			let body_data = format!("do=search&subaction=search&titleonly=3&story={}", search_value);
+
+			request = Request::new(url.as_str(), HttpMethod::Post)
+						.body(body_data.as_bytes())
+						.header("Referer", "https://manga.in.ua");
+		}
+		// genre and status -> ignore status
+		else if search_value.is_empty() && !genre_value.is_empty() && !status_value.is_empty(){
+			let url = format!("{}/mangas/{}/page/{}", base_url, genre_value, page);
+			request = Request::new(url.as_str(), HttpMethod::Get);
+			
+			is_cover_data_src = true;
+		}
+		// only search
+		else if !search_value.is_empty() && genre_value.is_empty() && status_value.is_empty(){
+			let url = format!("{}/index.php?do=search", base_url);
+			
+			let body_data = format!("do=search&subaction=search&story={}&search_start={}", search_value, page);
+
+			request = Request::new(url.as_str(), HttpMethod::Post)
+						.body(body_data.as_bytes())
+						.header("Referer", "https://manga.in.ua");
+		}
+		// only genre
+		else if search_value.is_empty() && !genre_value.is_empty() && status_value.is_empty(){
+			let url = format!("{}/mangas/{}/page/{}", base_url, genre_value, page);
+			request = Request::new(url.as_str(), HttpMethod::Get);
+			
+			is_cover_data_src = true;
+		}
+		// only status
+		else if search_value.is_empty() && genre_value.is_empty() && !status_value.is_empty(){
+			let url = format!("{}/{}/page/{}", base_url, status_value, page);
+			request = Request::new(url.as_str(), HttpMethod::Get);
+
+			is_cover_data_src = true;
+		}
+		// should not got here. just in case
+		else {
+			let url = format!("{}/page/{}", base_url, page);
+			request = Request::new(url.as_str(), HttpMethod::Get);
+
+			is_cover_data_src = true;
+		}
+
+		let html = request.html();
+
+		for result in html.select(".main .item").array() {
+			let res_node = result.as_node();
+
+			let card = res_node.select(".card--big .card__content");
+
+			let title = card.select(".card__title a").text().read();
+			
+			let href = card.select(".card__title a").attr("href").read();	
+
+
+			let cover : String = if is_cover_data_src {
+				res_node.select(".card--big img").attr("abs:data-src").read()
+			}
+			else{
+				res_node.select(".card--big img").attr("abs:src").read()
+			};
+
+			let mut categories: Vec<String> = Vec::new();
+			for categ_res in res_node.select(".card__category a").array(){
+				let categ_node = categ_res.as_node();
+				let categ_name = categ_node.text().read(); 
+				
+				categories.push(categ_name);
+			}
+			
+			let mut desc = String::new();
+			let mut is_nsfw : MangaContentRating = MangaContentRating::Safe;
+			for info in card.select(".card__list li").array(){
+				let info_node = info.as_node();
+				desc = info_node.text().read();
+				is_nsfw = helper::is_nsfw(desc.clone());
+				continue; // need only first element. is there better solution?
```

Comment:
?
you're continuing at the end of the block so it's doing nothing. if you just want the first element, use `.get(0)`.

Reply:
Thanks